k8s/envoy-opa/README.md

```diff
@@ -0,0 +1,346 @@
+
+```
+{
+  "decision_id": "207b7b54-0ec0-4ffb-a531-c86a9f05c38d",
+  "input": {
+    "attributes": {
+      ...
+      "request": {
+        "http": {
+          "headers": {
+            ":authority": "localhost:3003",
+            ":method": "GET",
+            ":path": "/profiles/2",
+            "accept-encoding": "gzip",
+            "content-length": "0",
+            "user-agent": "Go-http-client/1.1",
+            "x-forwarded-client-cert": "By=spiffe://example.org/ns/default/sa/default/backend;Hash=a9317919875e178ce6d61eaa023490a2091299753ca5cd01d5323e40696d690b;URI=spiffe://example.org/ns/default/sa/default/frontend",
+            "x-forwarded-proto": "http",
+            "x-request-id": "e0939bcf-8beb-4910-a980-be0468ec023f"
+          },
+          "method": "GET",
+          "path": "/profiles/2",
+          "protocol": "HTTP/1.1"
+        },
+      ...
+      },
+      ...
+    },
+   ...
+  },
+
+  "msg": "Decision Log",
+  "query": "data.envoy.authz.allow",
+  "requested_by": "",
+  "result": true,
+  "time": "2020-06-11T18:58:47Z",
+  "timestamp": "2020-06-11T18:58:47.840319148Z",
+  "type": "openpolicyagent.org/decision_logs"
+}
+```
+
+Note that the `URI` part of the `x-forwarded-client-cert` header contains the SPIFFE ID defined for the `frontend` service and it matches the SPIFFE ID defined in the rego policy configured for the OPA Agent. Request's path and method also match the rule, that is why the results for the decision is `true` and the request is allowed to pass through the filter and reach the `backend` service.
+
+## Testing Invalid Requests
+
+On the other hand, following the same steps we can confirm that a request that does not satisfy the policy prevent the associated data from being displayed. In this case the `frontend-2` service is not able to talk to the `backend` service because its SPIFFE ID does not satisfy the policy at the OPA Agent.
+As a result, when you connect to the URL for the `frontend-2` service (e.g. `http://35.222.190.182:3002`), the browser only displays the title without any account details.
+
+![Frontend-2-no-details][frontend-2-view-no-details]
+
+[frontend-2-view-no-details]: images/frontend-2_view_no_details.png "Frontend-2 view no details account"
+
+The reader can verify the decision made by OPA using the same script `scripts/backend-opa-logs.sh` as performed in the previous section. A similar log entry is available for the `frontend-2` service but with result equal to `false` due to the SPIFFE ID mismatch.
+
+## Retesting frontend-2 with a New Policy
+
+Let's update the rego policy to match the SPIFFE ID of the `frontend-2` and test again. There is a Bash script that we can leverage to complete this task. Once executed, it will open the editor defined by your `KUBE_EDITOR`, or `EDITOR` environment variables, or fall back to 'vi' for Linux or 'notepad' for Windows.
```

Comment:
```suggestion
Let's update the Rego policy to match the SPIFFE ID of the `frontend-2` and test again. There is a Bash script that we can leverage to complete this task. Once executed, it will open the editor defined by your `KUBE_EDITOR`, or `EDITOR` environment variables, or fall back to `vi` for Linux or Notepad for Windows.
```